plugin/completion.py

```diff
@@ -102,26 +102,27 @@ def translated_regions(self, edit_region: sublime.Region) -> Generator[sublime.R
 
     def epilogue(self, item: CompletionItem, session_name: str) -> None:
         session = self.session_by_name(session_name, 'completionProvider.resolveProvider')
+
+        def resolve_on_main_thread(item: CompletionItem, session_name: str) -> None:
+            sublime.set_timeout(lambda: self.on_resolved(item, session_name))
+
         if session:
             request = Request.resolveCompletionItem(item, self.view)
-            session.send_request_async(request, lambda response: self.on_resolved(response, session_name))
+            session.send_request_async(request, lambda response: resolve_on_main_thread(response, session_name))
         else:
-            self.on_resolved(item, session_name)
+            resolve_on_main_thread(item, session_name)
```

Comment:
Isn't this one already on the main thread and doesn't need to go through set_timeout?

Reply:
Correct